tar/bsdtar.c

```diff
@@ -143,7 +201,7 @@ main(int argc, char **argv)
 	 * for ease of cleanup.
 	 */
 	bsdtar = &bsdtar_storage;
-	memset(bsdtar, 0, sizeof(*bsdtar));
+	bsdtar_init(bsdtar);
```

Comment:
Might be cleaner to make bsdtar_init return a pointer so that this becomes

    bsdtar = bsdtar_init();